### PR TITLE
GCS migration, packaging fixes

### LINKED ISSUE #187: MIgrate from S3 to GCS
Depends on Airflow jobs migration.

### PR DESCRIPTION
**Pull request by eu9ene**
Switched from S3 to GCS.

Unfortunately, I had to refactor a lot of things to make it work in the Ensemble Spark job. Currently, some old version of taar package (before Redis cache was introduced) used there which doesn't make any sense for data flywheel.

Things refactored:
- data loader (cache) is split into two: redis one and local to be used in Spark job
- settings to have proper defaults for Spark job
- contexts - some immature dependency injection to provide different configurations for app and package

Closes #187

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,8 +88,10 @@ container and run the test suite inside the container.
 
 TAAR uses miniconda and a enviroment.yml file to manage versioning.
 
-To update versions, edit the enviroment.yml with the new dependency
-you need.  If you are unfamiliar with using conda, see the [official
+To update versions, edit the `enviroment.yml` with the new dependency
```
Comment (jasonthomas) [inline README.md:91]:
```suggestion
To update versions, edit the `environment.yml` with the new dependency
```

File: taar/settings.py [hunk 2]
```diff
@@ -5,79 +5,81 @@
...
 
-PYTHON_LOG_LEVEL = config("PYTHON_LOG_LEVEL", "INFO")
 
-REDIS_HOST = config("REDIS_HOST", "localhost", cast=str)
-REDIS_PORT = config("REDIS_PORT", 6379, cast=int)
+class AppSettings:
+    PYTHON_LOG_LEVEL = config("PYTHON_LOG_LEVEL", "INFO")
+    STATSD_HOST = config("STATSD_HOST", default="localhost", cast=str)
+    STATSD_PORT = config("STATSD_PORT", default=8125, cast=int)
+    NO_REDIS = config("NO_REDIS", False, cast=bool)
```
Comment (jasonthomas) [inline taar/settings.py:13]: Can we change this to `ENABLE_REDIS` instead? I think it is easier to read.

Comment (eu9ene) [reply taar/settings.py]: is `DISABLE_REDIS` ok? I assume that default behaviour is with redis enabled and we disable it for some specific cases, so `DISABLE_` feels more appropriate

File: README.md (hunk 1 above)
Comment (eu9ene) [reply README.md]: done

File: taar/recommenders/cache.py [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+
+# TAAR: ensemble weights
+
+ENSEMBLE_WEIGHTS = "taar_ensemble_weights|"
+
+# TAAR: whitelist data
+WHITELIST_DATA = "taar_whitelist_data|"
+
+
+class TAARCache(ITAARCache):
```
Comment (eu9ene) [inline taar/recommenders/cache.py:56]: This thing is definitely way more complex than it needs to be, but I tried to touch as little as possible to not break things.

File: taar/settings.py (hunk 2 above)
Comment (jasonthomas) [reply taar/settings.py]: Works for me

**Review by jasonthomas — APPROVED** [review]

**Pull request merged**